var/spack/repos/builtin/packages/cpmd/package.py

```diff
@@ -3,73 +3,209 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+
+import os
+
 from spack.package import *
 
 
-class Cpmd(MakefilePackage):
+class Cpmd(AutotoolsPackage):
     """The CPMD code is a parallelized plane wave / pseudopotential
-    implementation of Density Functional Theory, particularly
-    designed for ab-initio molecular dynamics."""
+    implementation of Density Functional Theory, particularly designed
+    for ab-initio molecular dynamics."""
 
-    homepage = "https://www.cpmd.org/wordpress/"
+    homepage = "https://github.com/CPMD-code"
     url = "https://github.com/CPMD-code/CPMD/archive/refs/tags/4.3.tar.gz"
+    git = "https://github.com/CPMD-code/CPMD.git"
+    maintainers = ["g-mathias"]
 
-    version("4.3", sha256="e0290f9da0d255f90a612e60662b14a97ca53003f89073c6af84fa7bc8739f65")
-
-    variant("omp", description="Enables the use of OMP instructions", default=False)
-    variant("mpi", description="Build with MPI support", default=False)
+    version(
+        "4.3",
+        default=True,
+        sha256="e0290f9da0d255f90a612e60662b14a97ca53003f89073c6af84fa7bc8739f65",
+    )
+    version("develop", branch="main")
 
+    depends_on("blas")
     depends_on("lapack")
-    depends_on("mpi", when="+mpi")
-
-    conflicts("^openblas threads=none", when="+omp")
-    conflicts("^openblas threads=pthreads", when="+omp")
-
-    def edit(self, spec, prefix):
-        # patch configure file
-        cbase = "LINUX-GFORTRAN"
-        cp = FileFilter(join_path("configure", cbase))
-        # Compilers
-        if spec.satisfies("+mpi"):
-            fc = spec["mpi"].mpifc
-            cc = spec["mpi"].mpicc
-        else:
-            fc = spack_fc
-            cc = spack_cc
 
-        cp.filter("FC=.+", "FC='{0}'".format(fc))
-        cp.filter("CC=.+", "CC='{0}'".format(cc))
-        cp.filter("LD=.+", "LD='{0}'".format(fc))
+    variant("mpi", default=True, description="Enable MPI support")
+    variant("debug", default=False, description="keep symbols for debugging")
+
+    with when("+mpi"):
+        depends_on("mpi@2:")
+    variant("openmp", default=True, description="Enable OpenMP support")
+    conflicts("^openblas threads=none", when="+openmp")
+    conflicts("^openblas threads=pthreads", when="+openmp")
+    extraflags = [
+        "ALPHALINUX",
+        "ALTIX",
+        "BG",
+        "CHECKS_IEEE_ARITHMETIC",
+        "CUDA",
+        "EREG",
+        "ES",
+        "FFT_HASNT_THREADED_COPIES",
+        "FFT_HAS_LOW_LEVEL_TIMERS",
```

Comment:
I'm confused about this giant array. Isn't the build system supposed to set these?